buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy

```diff
@@ -162,7 +164,25 @@ class NodeInfo {
             args.add("${esScript}")
         }
 
-        env = ['JAVA_HOME': project.runtimeJavaHome]
+        final String javaHome
+        final Map<Integer, JavaVersion> javaVersions = project.javaVersions
+        if (Version.fromString(nodeVersion).before("6.2.0")) {
```

Comment:
Can we change nodeVersion so it is a Version instead of String so we don't have to parse many times during configuration of every node in gradle?

Reply:
Sure; I will address this in a follow-up PR.

Reply:
I opened #29515.